tubular/dates.py

```diff
@@ -663,3 +665,152 @@ def transform(self, X):
         X[self.new_column_name] = lower_comparison & upper_comparison
 
         return X
+
+
+class DatetimeSinusoidCalculator(BaseTransformer):
+
+    """
+    Transformer to derive a feature in a dataframe by calculating the
```

Comment:
Looking at this transformer again, I think it seems a bit inconsistent that you can specify multiple input columns but only one unit or period - might be a bit much for now but could be worth adding a refactor to the backlog (e.g. to dictionaries or lists that are assumed to have matching inputs)